master/internal/api_trials.go

```diff
@@ -753,15 +753,28 @@ func (a *apiServer) GetTrialWorkloads(ctx context.Context, req *apiv1.GetTrialWo
 		limit = ptrs.Ptr[int32](-1)
 	}
 
+	metricsSortCode := "total_batches"
 	sortCode := "total_batches"
 	if req.SortKey != "" && req.SortKey != "batches" {
-		sortCode = fmt.Sprintf("sort_metrics->'avg_metrics'->>'%s'",
+		metricsSortCode = fmt.Sprintf("(metrics->'avg_metrics'->>'%s')::float",
+			strings.ReplaceAll(req.SortKey, "'", ""))
+		sortCode = fmt.Sprintf("(sort_metrics->'avg_metrics'->>'%s')::float",
```

Comment:
I'm regretting creating `sort_metrics` in the query now. It's only used to combine t.metrics/v.metrics for the sort and isn't returned to the API. You could revert it to 'metrics' and no longer need two names.